### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Retsediv**

### CODE REVIEW

File: hydra_lsp/intel.py [hunk 1]
```diff
@@ -2,26 +2,59 @@
...
+    to_markdown_content,
+    yaml_get_identifier,
+    yaml_get_variable_name,
+)
 
 logger = logging.getLogger(__name__)
 
 
-# TODO: this class has a lot of duplicated code
+def intel(feature: str, context_required: bool = True):
```
Comment (liepieshov) [inline hydra_lsp/intel.py:22]: if you want to nicely annotate wrappers, you can use TypeVar and ParamSpec

File: hydra_lsp/intel.py [hunk 2]
```diff
@@ -62,55 +85,31 @@ def get_hover(
         s = json.dumps({key: value}, indent=2)[1:-1]
```
Comment (liepieshov) [inline hydra_lsp/intel.py:85]: we can also use pprint here to format a nice string

File: pyproject.toml [hunk 3]
```diff
@@ -37,6 +37,15 @@ hydra-lsp = "hydra_lsp.__main__:main"
...
 build-backend = "poetry.core.masonry.api"
 
+[tool.black]
+line-length = 88
+target-version = ['py38', 'py39', 'py310', 'py311']
+
+[tool.isort]
+multi_line_output = 3
+line_length = 88
+include_trailing_comma = true
```
Comment (liepieshov) [inline pyproject.toml:47]: cool

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.10", "3.11"]
```
[the remark this answers is not in the document]
Comment (Retsediv) [reply .github/workflows/test.yml:16]: I guess there is little point in supporting 3.8/3.9 given how old they are. Also, this tool will be used inside editors and python version is not going to be a problem.

[the remark this answers is not in the document]
Comment (liepieshov) [reply .github/workflows/test.yml:16]: Second point makes much more sense.

**Review by liepieshov — APPROVED** [review]

**Pull request merged**